app/controllers/instances_controller.rb

```diff
@@ -340,4 +347,30 @@ def find_a_parent(parent)
     end
     @working_draft.tree_version_elements.order(tree_element_id: "asc").first
   end
+
+  def find_or_initialize_profile_items(product_id, instance_id)
+    # TODO: Optimise this method!! 
```

Comment:
This seems to be an expensive query as it gets the product configs and loop through it and checks for available profile item..I was thinking if we change the behaviour of the form, e.g show a dropdown of possible profile item  and only show profile items that has been added, we can reduce the query

Reply:
Yes, let's keep an eye on this as the design evolves.